lib/presentation/ui/home/home_page.dart

```diff
@@ -239,32 +238,64 @@ class RecentListWidget extends StatelessWidget {
  * and a project for this open-source project
  */
 class MapWidget extends StatelessWidget {
+  final HomeBloc bloc;
+  MapWidget(this.bloc);
+
   final MapController mapController = MapController();
   final LatLng _center = LatLng(39.806251, -8.088591);
 
   @override
   Widget build(BuildContext context) {
     return Stack(
       children: <Widget>[
-        FlutterMap(
-          mapController: mapController,
-          options: MapOptions(
-            center: _center,
-            zoom: 7.0,
-            minZoom: 1.0,
-            maxZoom: 20.0,
-          ),
-          layers: [
-            TileLayerOptions(
-              urlTemplate: MAPBOX_URL_TEMPLATE,
-              additionalOptions: {
-                'accessToken': MAPBOX_ACCESS_TOKEN,
-                'id': 'mapbox.streets',
-              },
-            ),
-          ],
-        ),
+        StreamBuilder<List<OccurrenceModel>>(
+            stream: bloc.mockDataStream,
+            builder: (context, snapshot) {
+              if (!snapshot.hasData) {
```

Comment:
If there is a problem with the internet connection or if this takes a long time, then we are not showing a map here.

Why not show always the map and have that text above it OR a CircularProgressIndicator that shows that data is loading?